src/app/Library/CrudPanel/Traits/Create.php

```diff
@@ -169,14 +171,111 @@ private function createRelationsForItem($item, $formattedRelations)
                 }
             } elseif ($relation instanceof HasOne || $relation instanceof MorphOne) {
                 $modelInstance = $relation->updateOrCreate([], $relationDetails['values']);
+            } elseif ($relation instanceof HasMany || $relation instanceof MorphMany) {
+                $relation_values = $relationDetails['values'][$relationMethod];
+                // if relation values are null we can only attach, also we check if we sent a single dimensional array [1,2,3], or an array of arrays: [[1][2][3]]
+                // if is as single dimensional array we can only attach.
+                if ($relation_values === null || count($relation_values) == count($relation_values, COUNT_RECURSIVE)) {
+                    $this->attachManyRelation($item, $relation, $relationMethod, $relationDetails, $relation_values);
+                } else {
+                    $this->createManyEntries($item, $relation, $relationMethod, $relationDetails);
+                }
             }
-
             if (isset($relationDetails['relations'])) {
                 $this->createRelationsForItem($modelInstance, ['relations' => $relationDetails['relations']]);
             }
         }
     }
 
+    /**
+     * When using the HasMany/MorphMany relations as selectable elements we use this function to "mimic-sync" in those relations.
+     * Since HasMany/MorphMany does not have the `sync` method, we manually re-create it.
+     * Here we add the entries that developer added and remove the ones that are not in the list.
+     * This removal process happens with the following rules:
+     * - by default Backpack will behave like a `sync` from M-M relations: it deletes previous entries and add only the current ones.
+     * - `force_delete` is configurable in the field, it's `true` by default. When false, if connecting column is nullable instead of deleting the row we set the column to null.
+     * - `fallback_id` could be provided. In this case instead of deleting we set the connecting key to whatever developer gives us.
```

Comment:
How will `fallback_id` play along with Laravel's `withDefault()` and the DB column `default`?

```php
// let's remember the dev might have done this to have
// an author called "Unknown Author" with id 1022
public function author()
{
    return $this->belongsTo('App\Author')->withDefault(1022);
}

// or inside their migration they might have set a `default` on the column
$table->integer('author_id')->default(1022);
```

From what I understand, the `fallback_id` will make Backpack set the value to that, so it won't even reach `withDefault()` or the `default` from the DB, is that correct? If so, that's great, I think that's the expected behavior.

Reply:
But if `fallback_id` is NOT specified, I think what should happen is this, in this order:
- if column is `NOT NULL`, delete 🛑
- if column is `nullable`
    - if a default has been set on the `belongsTo` relationship, set it to that;
    - if a default has been set on the db column, set it to that;
    - if no default on the db column... it's the same as setting the default to `null`, so set it to that;